core/buffer.lua

```diff
@@ -357,11 +389,14 @@ text.
 function reduxbuffer:add_text(text, style, command, indicator)
   text = tostring(text)
   local insert_pos = self.target.current_pos
+  local saved_buf = buffer
+  _G.buffer = self.target
```

Comment:
It's a horrible hack, isn't it?  It's been a while since I looked at it, but I think it's to do with the fact that the command entry buffer is special; when it's active, `_G.buffer` still points at the focused "normal" buffer.  So this would be to fool all the called code which expects to do things with `buffer`, to work on the command entry buffer instead.